chore(nc-gui): cleanup types and unused code

bcakmakoglu opened the pull request:
## Change Summary

* Add `useRoles` composable
* Reduce the amount of computed variables generated through `usePermission` by implementing `useInjectionState`
* Remove duplicating roles object on each permission check
   * Check preview permissions separately
* Cleanup permission types and role types
* Use enums for role names
* Remove unused code
* Remove unused comment
* Exclude `components.d.ts` from eslint

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [x] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [x] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

wingkwong commented:
Sorry. Just realise that it's a draft. Will take a look again when it's ready for review.

bcakmakoglu commented:
@wingkwong Yeah I reverted from PR to draft because I noticed some problems with `useState` and the project permissions which made it impossible to use a single computed prop.

I refactored it, so now it should work as expected and consume a bit less memory because we keep a single instead of multiple states.

But let's wait for CI/CD 😄

pranavxc reviewed (commented):
Please do a rebase and push.

pranavxc commented on packages/nc-gui/composables/useRoles/index.ts:
```diff
@@ -0,0 +1,65 @@
+        {},
+        {
+          headers: {
+            'xc-shared-base-id': projectId,
```
The `projectId` and `shared-base-id` are not the same, It's a unique UUID mapped to the project. 

The above line needs to be replaced with `'xc-shared-base-id': route.params.projectId`
